webapp/tests/mocha/unit/translator.spec.js

```diff
@@ -0,0 +1,65 @@
+const BootstrapTranslator = require('../../../src/js/bootstrapper/translator'),
+      { expect } = require('chai');
+
+const translationData = {
+  en: {
+    LOAD_ASSETS:   'Loading assets!',
+    FETCH_INFO:    x => `Fetching info (${x} docs)!`,
+    EN_ONLY:       'Yay!',
+  },
+  es: {
+    LOAD_ASSETS:   'Cargando activos!',
+    FETCH_INFO:    x => `Descarga de datos (${x} documentos)!`,
+    ES_ONLY:      'Vamos!',
+  },
+};
+
+describe('Bootstrap Translator', () => {
+  it('Translate LOAD_ASSETS to English', () => {
+    const translator = new BootstrapTranslator('en', translationData);
+    expect(translator.translate('LOAD_ASSETS')).to.eq(translationData.en.LOAD_ASSETS);
+  });
+
+  it('Translate defaults and setLocale to English', () => {
+    const translator = new BootstrapTranslator(undefined, translationData);
+    expect(translator.translate('LOAD_ASSETS')).to.eq(translationData.en.LOAD_ASSETS);
+    translator.setLocale('es');
+    expect(translator.translate('LOAD_ASSETS')).to.eq(translationData.es.LOAD_ASSETS);
+  });
+
+  it('Translate to Spanish', () => {
+    const translator = new BootstrapTranslator('es', translationData);
+    expect(translator.translate('LOAD_ASSETS')).to.eq(translationData.es.LOAD_ASSETS);
+  });
+
+  it('Translate with embedded argument to Spanish', () => {
+    const translator = new BootstrapTranslator('es', translationData);
+    expect(translator.translate('FETCH_INFO', 35)).to.eq('Descarga de datos (35 documentos)!');
+  });
+
+  it('Missing translation falls back to English', () => {
+    const translator = new BootstrapTranslator('es', translationData);
+    expect(translator.translate('EN_ONLY')).to.eq(translationData.en.EN_ONLY);
+  });
+
+  it('Static definitions', () => {
```

Comment:
Does this test assert that keys are available as eponymous constants?  How is the feature used?

Reply:
`translator.translate(BootstrapTranslator.TRY_AGAIN)`

It's my attempt to have an `enum` - but arguably serves no purpose.